crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -238,25 +258,65 @@ impl<S> P2PStream<S> {
         &self.shared_capability
     }
 
+    pub fn new_capability(
+        &mut self,
+        capability: Capability,
```

Comment:
is this even possible? I thought capabilities (and their offsets) are negotiated during the rlpx handshake?

Reply:
true, `SharedCapability` is already set for the stream, to be fair it was a placeholder, wip pr. how do you have in mind to code multiple shared capabilities, something like `pub struct SharedCapabilitySet(Vec<SharedCapability>)` and setting that type in a p2pstream instead than `SharedCapability`?

Reply:
I think that would work!